src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -275,12 +275,12 @@ private void OnGetDataMessageReceived(InvPayload payload)
         private void OnGetHeadersMessageReceived(GetBlockByIndexPayload payload)
         {
             DataCache snapshot = Blockchain.Singleton.View;
-            if (payload.IndexStart > Blockchain.Singleton.HeaderHeight(snapshot)) return;
+            if (payload.IndexStart > NativeContract.Ledger.CurrentIndex(snapshot)) return;
```

Comment:
Shouldn't header cached be considered in this scenario?

Reply:
We don't need to send the headers in the cache to other nodes. Because we only cache 10,000 headers, there is no need to make this complicated.